hardware_interface/src/component_parser.cpp

```diff
@@ -611,6 +629,96 @@ std::vector<HardwareInfo> parse_control_resources_from_urdf(const std::string &
     ros2_control_it = ros2_control_it->NextSiblingElement(kROS2ControlTag);
   }
 
+  // parse full URDF for mimic options
+  urdf::Model model;
+  if (!model.initString(urdf))
+  {
+    throw std::runtime_error("Failed to parse URDF file");
+  }
+  for (auto & hw_info : hardware_info)
+  {
+    for (auto i = 0u; i < hw_info.joints.size(); ++i)
+    {
+      const auto & joint = hw_info.joints.at(i);
+
+      // Search for mimic joints defined in ros2_control tag (deprecated)
+      // TODO(christophfroehlich) delete deprecated config with ROS-J
+      if (joint.parameters.find("mimic") != joint.parameters.cend())
+      {
+        std::cerr << "Warning: Mimic joints defined in ros2_control tag are deprecated. "
+                  << "Please define mimic joints in URDF." << std::endl;
+        const auto mimicked_joint_it = std::find_if(
+          hw_info.joints.begin(), hw_info.joints.end(),
+          [&mimicked_joint =
+             joint.parameters.at("mimic")](const hardware_interface::ComponentInfo & joint_info)
+          { return joint_info.name == mimicked_joint; });
+        if (mimicked_joint_it == hw_info.joints.cend())
+        {
+          throw std::runtime_error(
+            std::string("Mimicked joint '") + joint.parameters.at("mimic") + "' not found");
+        }
+        hardware_interface::MimicJoint mimic_joint;
+        mimic_joint.joint_index = i;
+        mimic_joint.multiplier = 1.0;
+        mimic_joint.offset = 0.0;
+        mimic_joint.mimicked_joint_index = std::distance(hw_info.joints.begin(), mimicked_joint_it);
+        auto param_it = joint.parameters.find("multiplier");
+        if (param_it != joint.parameters.end())
+        {
+          mimic_joint.multiplier = hardware_interface::stod(joint.parameters.at("multiplier"));
+        }
+        param_it = joint.parameters.find("offset");
+        if (param_it != joint.parameters.end())
+        {
+          mimic_joint.offset = hardware_interface::stod(joint.parameters.at("offset"));
+        }
+        hw_info.mimic_joints.push_back(mimic_joint);
+      }
+      else
+      {
+        auto urdf_joint = model.getJoint(joint.name);
+        if (!urdf_joint)
+        {
+          throw std::runtime_error("Joint " + joint.name + " not found in URDF");
+        }
+        if (!urdf_joint->mimic && joint.is_mimic == MimicAttribute::TRUE)
+        {
+          throw std::runtime_error(
+            "Joint '" + std::string(joint.name) + "' has no mimic information in the URDF.");
+        }
+        if (urdf_joint->mimic && joint.is_mimic != MimicAttribute::FALSE)
+        {
+          if (joint.command_interfaces.size() > 0)
+          {
+            throw std::runtime_error(
+              "Joint '" + std::string(joint.name) +
+              "' has mimic attribute not set to false: Mimic joints cannot have command "
+              "interfaces.");
```

Comment:
```suggestion
            throw std::runtime_error(
              "Joint '" + std::string(joint.name) +
              "' has mimic attribute set to true: Mimic joints cannot have command "
              "interfaces.");
```

Reply:
No that's not the same. If it is not explicitly set to false, then it is default `true` and the error is thrown.

Reply:
Gotcha!